src/msg/async/Timeout.h

```diff
@@ -28,7 +28,8 @@
 constexpr int
 TimeoutToMilliseconds(const struct timeval &tv) noexcept
 {
-  return tv.tv_sec * 1000 + tv.tv_usec / 1000;
+  /* round up to the next millisecond so we don't wake up too early */
+  return tv.tv_sec * 1000 + (tv.tv_usec + 999) / 1000;
```

Comment:
nit: `include/intarith.h` already offers `round_up_to()`.

Reply:
Oh, nice - but `round_to_up()` is bugged: it explicitly casts to unsigned, for whatever reason. Worse: it (implicitly) casts to unsigned *after* the division. Which makes no sense.
I changed it, and the bad cast has no effect because it's immediately cast back to `int` which is the type needed by `epoll_wait()`.